Content/ContentTypeResolver/SingleSnippetSelectionResolver.php

```diff
@@ -0,0 +1,115 @@
+use Sulu\Bundle\SnippetBundle\Snippet\DefaultSnippetManagerInterface;
+use Sulu\Bundle\SnippetBundle\Snippet\WrongSnippetTypeException;
+use Sulu\Component\Content\Compat\PropertyInterface;
+use Sulu\Component\Content\Compat\Structure\SnippetBridge;
+use Sulu\Component\Content\Compat\Structure\StructureBridge;
+use Sulu\Component\Content\Mapper\ContentMapperInterface;
+
+class SingleSnippetSelectionResolver implements ContentTypeResolverInterface
+{
+    /**
+     * @var StructureResolverInterface
+     */
+    private $structureResolver;
+
+    /**
+     * @var ContentMapperInterface
+     */
+    private $contentMapper;
+
+    /**
+     * @var DefaultSnippetManagerInterface
+     */
+    private $defaultSnippetManager;
+
+    public function __construct(
+        ContentMapperInterface $contentMapper,
+        StructureResolverInterface $structureResolver,
+        DefaultSnippetManagerInterface $defaultSnippetManager
+    ) {
+        $this->contentMapper = $contentMapper;
+        $this->structureResolver = $structureResolver;
+        $this->defaultSnippetManager = $defaultSnippetManager;
+    }
+
+    public static function getContentType(): string
+    {
+        return 'single_snippet_selection';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function resolve($data, PropertyInterface $property, string $locale, array $attributes = []): ContentView
+    {
+        /** @var StructureBridge $structure */
+        $structure = $property->getStructure();
+        $webspaceKey = $structure->getWebspaceKey();
+        $shadowLocale = $structure->getIsShadow() ? $structure->getShadowBaseLanguage() : null;
+
+        $params = $property->getParams();
+        /** @var bool $loadExcerpt */
+        $loadExcerpt = isset($params['loadExcerpt']) ? $params['loadExcerpt']->getValue() : false;
+        /** @var string $defaultArea */
+        $defaultArea = isset($params['default']) ? $params['default']->getValue() : null;
+
+        $snippetId = $data ?? null;
+        if (empty($snippetId) && $defaultArea) {
+            $defaultSnippetId = $this->getDefaultSnippetId($webspaceKey, $defaultArea, $locale);
+            $snippetId = $defaultSnippetId ?: null;
+        }
+
+        if (null === $snippetId) {
+            return new ContentView(null);
+        }
+
+        /** @var SnippetBridge $snippet */
+        $snippet = $this->contentMapper->load($snippetId, $webspaceKey, $locale);
+
+        if (!$snippet->getHasTranslation() && null !== $shadowLocale) {
+            /** @var SnippetBridge $snippet */
+            $snippet = $this->contentMapper->load($snippetId, $webspaceKey, $shadowLocale);
+        }
+
+        $snippet->setIsShadow(null !== $shadowLocale);
+        $snippet->setShadowBaseLanguage($shadowLocale);
+
+        $resolvedSnippet = $this->structureResolver->resolve($snippet, $locale, $loadExcerpt);
+
+        return new ContentView($resolvedSnippet, ['id' => $data]);
```

Comment:
would you mind to adjust the existing `SnippetSelectionResolver` to use this format too? 🙂  (`['ids' => ...]`). then this would be consistent for the snippet selections

Reply:
Looks like we forget that when fixing #46 same should be done for the Page and SinglePageSelectionResolver. Then I think all should be consistent.

Reply:
i guess it would also make sense to use that format in case there is no value set 🙂 for example here:
https://github.com/sulu/SuluHeadlessBundle/blob/1bea2a10d52a1b062311fabb067dee71a422bb91/Content/ContentTypeResolver/SingleContactSelectionResolver.php#L49-L51

but we dont need to do this in this pull request

Reply:
sorry i'm not sure anymore :D should i change something more ?

Reply:
no - i will create a PR for the other resolvers 🙂